### DISCUSSION TITLE
How to configure random suffix length while using autonaming?

### DISCUSSION THREAD

**Discussion by TechWatching** [pulumi/pulumi#9125] [Pulumi in Practice] [answered]
As described in the [documentation](https://www.pulumi.com/docs/intro/concepts/resources/names/#autonaming), autonaming has a lot of benefits. However in practice when using it (especially with Azure), it often lead to exceed the resource's name length limit. 

Overriding auto-name is not a good solution. However, if it were possible to specify the random suffix length for a specific resource it would probably solve most of our naming issues. It is something that has been proposed [here](https://github.com/pulumi/pulumi/issues/1518#issuecomment-646594645) #1518.

Is it something that can be currently done? Is it something Pulumi could implement?

**Comment by Frassle** [comment, marked as answer]
> However in practice when using it (especially with Azure), it often lead to exceed the resource's name length limit.

That's a bug. The autonaming code should error if the user name plus the random characters exceeds the max support name for the given resource. 

This may be due to a problem in the autonaming code, and/or bugs in the resource definitions. If you could let us know what resources you've had this problem with (and what names) we can look into that.

> It is something that has been proposed https://github.com/pulumi/pulumi/issues/1518#issuecomment-646594645 https://github.com/pulumi/pulumi/issues/1518.

Currently there is no configuration of autonames from user programs. #1518 tracks adding some configurability here, suffix length may be a useful option even after fixing the length bugs mentioned above.

**Reply by TechWatching** [reply]
Thanks for your answer. The last resource that I had an issue with was the azure key vault.
Here was the error:
```
error: azure-native:keyvault:Vault resource 'kv-sqlbindings-dev' has a problem: 'vaultName' does not match expression '^[a-zA-Z0-9-]{3,24}$'
```
It is not very explicit, it did not show the real physical name, only the logical so I had to guess that autonaming was the problem.

**Reply by Frassle** [reply]
Thanks have raised https://github.com/pulumi/pulumi-azure-native/issues/1550
